x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/filters/filters.tsx

```diff
@@ -0,0 +1,282 @@
+    params: {
+      filters: column.params.filters.length > 0 ? column.params.filters : [defaultSearchQuery],
+    },
+  }),
+  onFieldChange: (oldColumn, indexPattern, field) => {
+    return {
+      ...oldColumn,
+      label: field.displayName,
+      sourceField: field.name,
+    };
+  },
+  paramEditor: ({ state, setState, currentColumn, layerId }) => {
+    const indexPattern = state.indexPatterns[state.layers[layerId].indexPatternId];
+
+    const filters = currentColumn.params.filters;
+
+    const onDragEnd = ({
+      source,
+      destination,
+    }: {
+      source?: DraggableLocation;
+      destination?: DraggableLocation;
+    }) => {
+      if (source && destination) {
+        const items = euiDragDropReorder(filters, source.index, destination.index);
+        setFilters(items);
+      }
+    };
+
+    const setFilters = (newFilters: Filter[]) =>
+      setState(
+        updateColumnParam({
+          state,
+          layerId,
+          currentColumn,
+          paramName: 'filters',
+          value: newFilters,
+        })
+      );
+
+    return (
+      <EuiForm>
+        <EuiFormRow
+          label={i18n.translate('xpack.lens.indexPattern.filters.queries', {
+            defaultMessage: 'Queries',
+          })}
+        >
+          <div>
+            <EuiDragDropContext onDragEnd={onDragEnd}>
+              <EuiDroppable droppableId="FILTERS_DROPPABLE_AREA" spacing="s">
+                {filters?.map((filter: Filter, idx: number) => {
+                  const { input, label } = filter;
+                  const id = `${JSON.stringify(input.query)}_${label}`;
+                  return (
+                    <EuiDraggable
+                      spacing="m"
+                      key={id}
+                      index={idx}
+                      draggableId={id}
+                      customDragHandle={true}
+                    >
+                      {(provided) => (
+                        <EuiPanel paddingSize="none">
+                          <EuiFlexGroup gutterSize="xs" alignItems="center" responsive={false}>
+                            <EuiFlexItem grow={false}>
+                              <div {...provided.dragHandleProps} className="lnsLayerPanel__dndGrab">
+                                <EuiIcon
+                                  type="grab"
+                                  aria-label={i18n.translate(
+                                    'xpack.lens.indexPattern.filters.grabIcon',
+                                    {
+                                      defaultMessage: 'Grab icon',
+                                    }
+                                  )}
+                                />
+                              </div>
+                            </EuiFlexItem>
+                            <EuiFlexItem
+                              grow={true}
+                              data-test-subj="indexPattern-filters-existingFilterContainer"
+                            >
```

Comment:
I can't tell if this is an EUI bug or my expectations being wrong, but there is a problem in the interaction when I begin dragging the popover button instead of the drop handle. Because it's not draggable, I start highlighting text in my browser, even though I was expecting a drag to start.

Reply:
In my opinion that's a correct behaviour. @cchaos what do you think?

Reply:
I can see where the confusion is because Lens, in general, (will) allow dragging fields and configurations initiating the drag from **anywhere** on the trigger (not just the grab icon). One of the behaviors that EUI setup for DnD is to be triggered from the grab icon because, yes, the text may still need be selectable (they might want to copy it). But EUI also provides the option to make the whole button moveable and remove the grab icon.

The main reason my mocks, and **most** of the other presentations for DnD, use the grab handle is to project the idea that these things can be re-ordered. Otherwise it just looks like a list of static panels.

So lets go ahead and do both, show the grab icon and make the whole panel draggable. Since the text is actually copyable from the popover, I'm less worried about that need in this situation.